core/Widget/WidgetConfig.php

```diff
@@ -366,4 +367,31 @@ public function isWide()
         return $this->isWide;
     }
 
+    /**
+     * Returns whether it makes sense to have more than one of this widget in a single dashboard. For
+     * most reports, eg "Visits Over Time", having more than one of them would be redundant. Some widgets,
+     * like the "single metric view" are generic and it would make sense to have more than one of them.
+     *
+     * Note: it is up to the widget metadata consumer to respect or ignore this property.
+     *
+     * @return bool
+     */
+    public function isReusable()
```

Comment:
Is that new setting really needed? IMO adding any widget multiple times can always make sense as you may want to add it to dashboard multiple times with different visualizations or sort orders etc. 

If we keep it: `reusable` is maybe not best wording maybe as it is pretty much only useful for dashboard in this is not quite clear. 

I'd even rather hard code this in the dashboard JS instead of adding this new setting.

Reply:
Personally, it be fine to always remove the "grey" color and show it in regular color (I presume @mattab  will want to keep this functionality). If that's not fine and we still need to show it grey, I would rather hard code it for this widget in the dashboard then adding this new API. Eg based on the "Generic" category or widget unique Id or something.

Reply:
👍

Reply:
It's not a 100% requirement and sounds good to hardcode it. It's quite useful to show the already-added-widgets differently, as people may forget the widget is already added in the bottom of the dashboard.